x-element.js

```diff
@@ -1207,6 +1223,34 @@ class Template {
     return reference;
   }
 
+  static #exhaustString(string, state) {
```

Comment:
Rather than use a stateless regular expression to look back across the string — we now walk through each piece of the string with a state object. Although a bit more cumbersome, it's more performant and more correct.